src/analysis/Environments/Environments.test.ts

```diff
@@ -916,6 +878,115 @@ describe('Environments Component', () => {
       } satisfies DataRefreshInfo);
     });
   });
+
+  describe('Workspaces - ', () => {
```

Comment:
Added a few test cases to cover workspace refresh behavior (I verified they fail as expected on the old code and pass now).